Formula/openmodelica.rb

```diff
@@ -0,0 +1,64 @@
+class Openmodelica < Formula
+  desc "Open-source modeling and simulation tool"
+  homepage "https://openmodelica.org/"
+  # GitHub's archives lack submodules, must pull:
+  url "https://github.com/OpenModelica/OpenModelica.git",
+    tag:      "v1.16.5",
+    revision: "11fcab4f2d6895f2db073572b2bff1a43177313f"
+  license "GPL-3.0-only"
+  head "https://github.com/OpenModelica/OpenModelica.git"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "cmake" => :build
+  depends_on "gcc" => :build # for gfortran
+  depends_on "gnu-sed" => :build
+  depends_on "libtool" => :build
+  depends_on "openjdk" => :build
+  depends_on "pkg-config" => :build
+
+  depends_on "boost"
+  depends_on "gettext"
+  depends_on "hdf5"
+  depends_on "hwloc"
+  depends_on "lp_solve"
+  depends_on "omniorb"
+  depends_on "openblas"
+  depends_on "qt@5"
+  depends_on "readline"
+  depends_on "sundials"
+
+  uses_from_macos "curl"
+  uses_from_macos "expat"
+  uses_from_macos "libffi"
+  uses_from_macos "ncurses"
+
+  def install
+    ENV.append_to_cflags "-I#{MacOS.sdk_path_if_needed}/usr/include/ffi"
+    args = %W[
+      --prefix=#{prefix}
+      --disable-debug
+      --disable-modelica3d
+      --with-cppruntime
+      --with-hwloc
+      --with-lapack=-lopenblas
+      --with-omlibrary=core
+      --with-omniORB
+    ]
+
+    system "autoconf"
+    system "./configure", *args
+    # omplot needs qt & OpenModelica #7240.
+    # omparser needs OpenModelica #7247
+    # omshell, omedit, omnotebook, omoptim need QTWebKit: #19391 & #19438
+    # omsens_qt fails with: "OMSens_Qt is not supported on MacOS"
+    system "make", "omc", "omlibrary-core", "omsimulator"
+    prefix.install Dir["build/*"]
+  end
+
+  test do
+    system "#{bin}/omc", "--version"
+    # assert_match "OMPlot", shell_output("#{bin}/OMPlot --help", 1) # help is the only nonblocking flag
+    system "#{bin}/OMSimulator", "--version"
```

Comment:
Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Reply:
I will think.  Given the complexity of their build system, I just wanted to test that linkage worked & the process loads.

Reply:
I now test that `omc` correctly parses a trivial `model` into a `class`.